policy/release_manager.md

```diff
@@ -0,0 +1,49 @@
+# Release manager role
+
+## Appointment
+
+A release manager (RM) is appointed for a specific upcoming release.
+
+A RM may and should step down at any time, when they cannot fulfill the associated obligations. In this case, a new RM is appointed.
+
+The appointment of an RM ends with the declaration of the respective release's end-of-life by the RM.
+
+## Responsibilities
+
+Immediately after the previous release was made, the RM takes over decision making regarding merges into the integration branch for all developments targeting the respective release. Such merges can be made as soon as, but no earlier, than the RM indicating approval.
+
+It is the RM's responsibility to act on any merge request in a timely manner.
+
+An RM is not expected to be an expert in all techniques, features, and parts of the code base. Consequently, an RM should seek feedback prior to approving merge requests whenever necessary.
+
+The RM declares the time of a feature freeze for the upcoming release.
+
+The RM decides on the length of the consolidation period following a feature freeze, and decides which kinds of changes are still acceptable.
+
+The RM decides on the number and timing of potential release candidates, and on the timing of the final release.
+
+After the initial release, the RM takes over the decision making for the respective maintenance branch in the same fashion as for the prior development towards the initial release. The RM decides on the nature of changes qualified for maintenance releases, and their timing.
```

Comment:
Just so that I grasp the concept fully: That means that if I am release manager for 0.17 I act on all PRs targeting ``master``, and after 0.17 release I act on all PRs targeting ``maint``, correct?

Reply:
Yes.

(Although the names of these branches are not enshrined here. This would be more appropriate for the description of the actual technical release process, rather than the description of the person role. I'd say these two should be different)

Reply:
I think then we should add information and authorization on making releases, right? Or is everything automated already, e.g., upload to PyPi?

Reply:
I cannot speak to automation -- I am largely unaware of it.

Can you give an example, what you mean by "information and authorization on making releases"?

I would also caution against an inflation of the RM role to including packaging.

From my POV a release is done by tagging a state in the Git repo. End of story.

Because we are presently using the GitHub platform, it makes sense to push that tag, and create a "GitHub release" too.

PyPi, to me, is a type of distribution, just like Conda, or Debian. I see updating all these channel in the responsibility of the respective package maintainers.

One could argue that an upload to PyPi is *that* simple, that a RM could also do it. But I think it is not a given implication.

Reply:
> From my POV a release is done by tagging a state in the Git repo. End of story.

Cool, thx for clarifying.

Reply:
Again a question regarding `master` and `maint`. I am using the names here because I do not find a clear definition that would, for example, define "development-branch", "integration-branch", or "maintenance-branch". Is [issue #4238](https://github.com/datalad/datalad/issues/4238) still governing the release process? (If so, should it not say in the last comment before 4.i: "merge maint into master"?)

If understand the process outlined in #4238 correctly, new features go into `master` and bug-fixes into  `maint`. So should an RM not focus on both branches? Or do we assume that bug-fixes are always prioritized, so no RM decision is needed w.r.t. bug-fixes?

Reply:
I am not responding to the first paragraph here. I think whether or not #4238 should (also) stay/become policy is a worthwhile discussion, but a separate one.

> So should an RM not focus on both branches?

No, there is a dedicated RM for in charge of each of them.

> Or do we assume that bug-fixes are always prioritized, so no RM decision is needed w.r.t. bug-fixes?

The RM for the upcoming release is also the assistant RM for the current release (see last section in the document). So procedurally, the decision making is happening on a team of two, or within a single person. I think this is not too complex.

But I want to emphasize again that this document is written in the spirit of a volunteer-driven project. So if a PR is opened for the upcoming release, and the RM for the current release thinks it better goes there (first), this person can *ask* for the change -- but there is no guarantee that such a change will be made by the volunteer contributor. Both RMs in charge  can then act in any which way they think is best (as it is already the case: rebase, or cherrypick to maint, etc).

Reply:
Thx for clarifying. I misunderstood the scope of release.